app/views/users/directions.js

```diff
@@ -0,0 +1,122 @@
+ <body>
+    <input id="origin-input" class="controls" type="text"
+        placeholder="Enter an origin location">
+
+    <input id="destination-input" class="controls" type="text"
+        placeholder="Enter a destination location">
+
+    <div id="mode-selector" class="controls">
+      <input type="radio" name="type" id="changemode-walking" checked="checked">
+      <label for="changemode-walking">Walking</label>
+
+      <input type="radio" name="type" id="changemode-transit">
+      <label for="changemode-transit">Transit</label>
+
+      <input type="radio" name="type" id="changemode-driving">
+      <label for="changemode-driving">Driving</label>
+    </div>
+
+    <div id="map"></div>
+
+    <script>
+function initMap() {
```

Comment:
Two things:
- This is an html file with embedded JS instead of a JS file. I think it's better practice to separate the form on lines 1 - 20 into an HTML file that requires this JS file.
- When you start defining script functions on line 22, those should be indented under the script tag, I think.